lms/djangoapps/shoppingcart/pdf.py

```diff
@@ -0,0 +1,444 @@
+        self.terms_conditions_text = microsite.get_value("PDF_RECEIPT_TERMS_AND_CONDITIONS", settings.PDF_RECEIPT_TERMS_AND_CONDITIONS)
+
+        # From Context
+        self.items_data = context['items_data']
+        self.id = context['id']
+        self.date = ModuleI18nService().strftime(context['date'], 'SHORT_DATE')
+        self.is_invoice = context['is_invoice']
+        self.total_cost = '{currency}{amount:.2f}'.format(currency=self.currency, amount=context['total_cost'])
+        self.payment_received = '{currency}{amount:.2f}'.format(currency=self.currency, amount=context['payment_received'])
+        self.balance = '{currency}{amount:.2f}'.format(currency=self.currency, amount=context['balance'])
+
+
+    def is_on_first_page(self):
+        """
+        Returns True if it's the first page of the pdf, False otherwise.
+        """
+        return self.pdf.current_page_count() == 1
+
+    def prepare_invoice_draw(self):
+        """
+        Setting up the pdf.
+        """
+        self.MARGIN = 15 * mm
+        self.page_width = letter[0]
+        self.page_height = letter[1]
+        self.min_clearance = 3 * mm
+
+        self.pdf = NumberedCanvas(self.file_buffer, pagesize=letter)
+
+    def generate_pdf(self, file_buffer):
+        """
+        Takes in a buffer and puts the generated pdf into that buffer.
+        """
+        self.file_buffer = file_buffer
+
+        self.prepare_invoice_draw()
+
+        self.draw_border()
+        y_pos = self.draw_logos()
+        self.second_page_available_height = y_pos - self.MARGIN - self.min_clearance
+        self.second_page_start_y_pos = y_pos
+
+        y_pos = self.draw_title(y_pos)
+        self.first_page_available_height = y_pos - self.MARGIN - self.min_clearance
+
+        y_pos = self.draw_course_info(y_pos)
+        y_pos = self.draw_totals(y_pos)
+        self.draw_footer(y_pos)
+
+        self.pdf.insert_page_break()
+        self.pdf.save()
+
+    def draw_border(self):
+        """
+        Draws a big border around the page leaving a margin of 15 mm on each side.
+        """
+        self.pdf.setStrokeColorRGB(0.5, 0.5, 0.5)
+        self.pdf.setLineWidth(0.353 * mm)
+
+        self.pdf.rect(self.MARGIN, self.MARGIN,
+                      self.page_width - (self.MARGIN * 2), self.page_height - (self.MARGIN * 2),
+                      stroke=True, fill=False)
+
+    @staticmethod
+    def load_image(img_path):
+        """
+        Loads an image given a path. If the path starts with "/static/images", then it looks
+        for the image in the lms/djangoapps/static/images folder, otherwise an absolute path is
+        assumed.
+        If the path points to an image file, it loads and returns the Image object.
+
+        """
+        if img_path.startswith("/static/images"):
+            folders_list_in_path = __file__.split(os.sep)[:-4]
+            folders_list_in_path.append(img_path)
+            abs_path = os.sep.join(folders_list_in_path)
```

Comment:
I believe I understand the path manipulation/calculation here, but is there a better way via some Django utility method to get an absolute filepath to the static directories?